Panic when ttn-lw-cli can't open listener for OAuth callback

**Summary:** 
<!-- Summarize the problem in a few sentences: -->

`ttn-lw-cli` panics when it can't open a listener for the OAuth callback.

**Steps to Reproduce:** 
<!-- How can we reproduce the problem? -->

Can't reproduce, but it's probably:

`$ ttn-lw-stack login` in an environment without permissions to open a listener

<!-- Please attach (or link to) configuration files if applicable -->

**What do you see now?**
<!-- Please attach (or link to) screenshots and logs if applicable -->

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xeb0c87]
 
goroutine 1 [running]:
go.thethings.network/lorawan-stack/cmd/ttn-lw-cli/commands.glob..func67(0x1abc740, 0x1b04e00, 0x0, 0x0, 0x0, 0x0)
               /home/travis/gopath/src/go.thethings.network/lorawan-stack/cmd/ttn-lw-cli/commands/login.go:73 +0x237
github.com/spf13/cobra.(*Command).execute(0x1abc740, 0x1b04e00, 0x0, 0x0, 0x1abc740, 0x1b04e00)
        /home/travis/gopath/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:762 +0x473
github.com/spf13/cobra.(*Command).ExecuteC(0x1abf460, 0xc00008c1a0, 0xc000561f88, 0x4056b0)
        /home/travis/gopath/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2fd
github.com/spf13/cobra.(*Command).Execute(0x1abf460, 0xc00008c1a0, 0x0)
        /home/travis/gopath/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800 +0x2b
main.main()
               /home/travis/gopath/src/go.thethings.network/lorawan-stack/cmd/ttn-lw-cli/main.go:27 +0x31
```

**What do you want to see instead?**
<!-- Please add some examples or mock-ups if applicable -->

An error.

**How do you propose to implement this?**
<!--
If unsure, add the discussion label and (temporarily) assign the expert
If you cannot assign people, please @mention the experts
-->

https://github.com/TheThingsNetwork/lorawan-stack/blob/master/cmd/ttn-lw-cli/commands/login.go#L73

We should always check errors first. Now there's an error, `lis` is `nil` and there's a panic.

**Environment:**
<!-- Your environment: OS/Browser/Gateway/Device/...? Versions? IDs/EUIs? -->

No permissions to start listener

**What can you do yourself and what do you need help with?**

Review

--- a/cmd/ttn-lw-cli/commands/login.go
+++ b/cmd/ttn-lw-cli/commands/login.go
@@ -70,10 +70,11 @@ var (
 				}))
 
 				lis, err := net.Listen("tcp", ":11885")
-				defer lis.Close()
 				if err != nil {
+					logger.WithError(err).Error("Could not listen for OAuth callback. Try re-running this command with --callback=false")
 					return err
 				}
+				defer lis.Close()
 				go http.Serve(lis, nil)
 			} else {
 				oauth2Config.RedirectURL = "/oauth/code"

